### ISSUE TITLE
MNT Reassess pandas and scikit-learn bounds

### ISSUE THREAD

**Issue by riedgar-ms** [fairlearn/fairlearn#1288]
The current bounds listed for `pandas` and `scikit-learn` in [requirements.txt](https://github.com/fairlearn/fairlearn/blob/main/requirements.txt) are laughably out of date, and we're getting test failures due to deprecation warnings.

**Comment by adrinjalali** [comment]
we should support like, last two years or something, not more.

**Comment by riedgar-ms** [comment]
So PR #1287 updates the SciKit-Learn lower bound (assuming we merge) but that still leaves pandas. And that has a lot of deprecation warnings

**Comment by adrinjalali** [comment]
yeah, and we need to make sure we work with latest pandas, which is very different. Also, we might need to do changes related to the upcoming numpy=2 release.

**Comment by riedgar-ms** [comment]
Looks like we also have issues with things being deprecated in `numpy` as well

**Comment by adrinjalali** [comment]
we could also do 3.9+ which would be 3.9, 3.10, 3.11, and the upcoming 3.12, which should be enough

**Comment by riedgar-ms** [comment]
Thoughts @fairlearn/fairlearn-maintainers ? I wouldn't object to bumping to >3.9, but this should probably be more widely considered.

**Comment by hildeweerts** [comment]
I also don't mind bumping to 3.9+ (and it's a good incentive for me to update my own installation, lol)

**Comment by riedgar-ms** [comment]
Thanks.... @MiroDudik @mmadaio @alliesaizan ?

**Comment by TamaraAtanasoska** [comment]
Maybe a year after removing python 3.8 is ok? :) Happy to make the change.

**Comment by adrinjalali** [comment]
Yeah I think we can remove now. Pythin 3.13 is out already, and we need to test for that.
